lib/src/settings.rs

```diff
@@ -100,13 +100,24 @@ pub struct SignSettings {
 impl SignSettings {
     /// Load the signing settings from the config.
     pub fn from_settings(settings: &UserSettings) -> Self {
-        let sign_all = settings.get_bool("signing.sign-all").unwrap_or(false);
+        // First try to get the explicit behavior setting
+        let behavior = match settings.get_string("signing.behavior").as_deref() {
+            Ok("drop") => SignBehavior::Drop,
+            Ok("keep") => SignBehavior::Keep,
+            Ok("own") => SignBehavior::Own,
+            Ok("force") => SignBehavior::Force,
```

Comment:
nit: Perhaps, this can be handled by `serde::Deserialize`. See `GraphStyle` for example.